lib/SVF-FE/SVFIRBuilder.cpp

```diff
@@ -1008,6 +1008,37 @@ const Value* SVFIRBuilder::getBaseValueForExtArg(const Value* V)
         if(totalidx == 0 && !SVFUtil::isa<StructType>(value->getType()))
             value = gep->getPointerOperand();
     }
+
+    // if the argument of memcpy is the result of an allocation (1) or a casted load instruction (2),
+    // further steps are necessary to find the correct base value
+    //
+    // (1)
+    // %call   = malloc 80
+    // %0      = bitcast i8* %call to %struct.A*
+    // %1      = bitcast %struct.B* %param to i8*
+    // call void memcpy(%call, %1, 80)
+    // 
+    // (2)
+    // %0 = bitcast %struct.A* %param to i8*
+    // %2 = bitcast %struct.B** %arrayidx to i8**
+    // %3 = load i8*, i8** %2
+    // call void @memcpy(%0, %3, 80)
+    LLVMContext &cxt = LLVMModuleSet::getLLVMModuleSet()->getContext();
+    if (value->getType() == PointerType::getInt8PtrTy(cxt)) {
+        // (1)
+        if (const CallBase* cb = SVFUtil::dyn_cast<CallBase>(value)) {
+            for (const User* user: cb->users()) {
+                if (const BitCastInst* bitCast = SVFUtil::dyn_cast<BitCastInst>(user))
+                    return bitCast;
+            }
```

Comment:
We should make sure the bitcast is the unique cast, otherwise, we may get the type from another cast not flowing into this memcpy?

Is the following what you want?
```````
if(SVFUtil::isHeapAllocExtCallViaRet(cb)){
  if(const Value* bitCast = getUniqueUseViaCastInst(cb))
     return bitCast;
}
````